neutone_sdk/gcn_1d.py

```diff
@@ -0,0 +1,381 @@
+        self.cond_dim = cond_dim
+        self.use_act = use_act
+        self.use_bias_in_conv = use_bias_in_conv
+
+        # Compute convolution channels and dilations
+        self.channels = [n_channels] * n_blocks
+        self.dilations = [dil_growth**idx for idx in range(n_blocks)]
+
+        # Blocks number is given by the number of elements in the channels list
+        self.n_blocks = len(self.channels)
+        assert len(self.dilations) == self.n_blocks
+
+        # Create a list of strides
+        self.strides = [1] * self.n_blocks
+
+        # Create a list of GCN blocks
+        self.blocks = nn.ModuleList()
+        block_out_ch = None
+
+        for idx, (curr_out_ch, dil, stride) in enumerate(
+            zip(self.channels, self.dilations, self.strides)
+        ):
+            if idx == 0:
+                block_in_ch = in_ch
+            else:
+                block_in_ch = block_out_ch
+            block_out_ch = curr_out_ch
+
+            self.blocks.append(
+                GCN1DBlock(
+                    block_in_ch,
+                    block_out_ch,
+                    self.kernel_size,
+                    dilation=dil,
+                    stride=stride,
+                    cond_dim=cond_dim,
+                    tfilm_block_size=tfilm_block_size,
+                    rnn_type=rnn_type,
+                    use_bias_in_conv=use_bias_in_conv,
+                )
+            )
+
+        # Output layer
+        self.out_net = nn.Conv1d(
+            self.channels[-1], out_ch, kernel_size=(1,), stride=(1,), bias=False
+        )
+
+        # Activation function
+        self.act = nn.Tanh()
+
+    def forward(self, x: Tensor, cond: Tensor) -> Tensor:
+        assert x.ndim == 3  # (batch_size, in_ch, samples)
+        if cond is not None:
+            assert cond.ndim == 2  # (batch_size, cond_dim)
+        for block in self.blocks:  # Apply GCN blocks
+            x = block(x, cond)
+        x = self.out_net(x)  # Apply output layer
+
+        if self.act is not None:
+            x = self.act(x)  # Apply tanh activation function
+        return x
+
+    def calc_receptive_field(self) -> int:
+        """Calculate the receptive field of the model.
+        The receptive field is the number of input samples that affect the output of a block.
+
+        The receptive field of the model is the sum of the receptive fields of all layers:
+        RF = 1 + \sum_{i=1}^{n}(kernel\_size_i - 1) \cdot dilation_i
+
+        i is the layer index, n is the number of layers.
+
+        Returns:
+            int: The receptive field of the model.
+        """
+        assert all(_ == 1 for _ in self.strides)  # TODO(cm): add support for dsTCN
+        assert self.dilations[0] == 1  # TODO(cm): add support for >1 starting dilation
+        rf = self.kernel_size
+        for dil in self.dilations[1:]:
+            rf = rf + ((self.kernel_size - 1) * dil)
+        return rf
+
+
+if __name__ == "__main__":
```

Comment:
As we mentioned in the meeting you can remove this once you're done testing

Reply:
Sure!